File: python/expreg_args/expreg.py
'''
_author   Jorge Eduardo Ascencio Espíndola
_version  v0.1
'''
import re, sys

def match(word):
    '''
    * Verifica que la cadena word cumpla con la expresión regular
    *
    * @param word   cadena a verificar
    *
    * @return True si cumple, False en otro caso
    '''
    expreg = re.compile('^[A-Z]([a-z]|ñ){1,5}$') # compila la expresión regular
    if expreg.match(word): # verifica que haya un match (regresa un objeto si cumple, None si no cumple)
        return True
    return False

if len(sys.argv) != 2:
    print('Recibe un solo argumento\nDebe ser de la forma [A-Z]([a-z]|ñ){1,5}')
else:
    if match(sys.argv[1]):
        print("Felicidades la cadena es valida!")
    else:
        print('La cadena no es correcta')
print('Cerrando programa ...')

File: java/expreg_args/Expreg.java
/**
 * @author   Jorge Eduardo Ascencio Espíndola
 * @version  v1.0
 */
public class Expreg{

    /**
     * Constructor
     */
    public Expreg(){
    }

    /**
     * Verifica si la cadena str cumple con la expresión regular
     *
     * @param str  cadena a evaluar
     *
     * return true si es valida, false en otro caso
     */
    public boolean match(String str){
	return str.matches("[A-Z]([a-z]|ñ){1,5}"); // verifica si cumple con la expresion regular
    }

    public static void main(String [] args){
	if(args.length != 1){
	    System.out.println("Recibe un solo argumento\nDebe ser de la forma");
	} else {
	    Expreg expr = new Expreg();
	    if(expr.match(args[0])){
		System.out.println("Felicidades la cadena es valida!");
	    } else {
		System.out.println("La cadena no es correcta");
	    }
	}
	System.out.println("Cerrando programa ...");
    }
}
